src/operator/tensor/dot-inl.h

```diff
@@ -231,6 +231,12 @@ inline bool DotForwardInferStorageType(const nnvm::NodeAttrs& attrs,
     dispatched = storage_type_assign(&out_stype, kDefaultStorage,
                                      dispatch_mode, DispatchMode::kFComputeEx);
   }
+  if (!dispatched && lhs_stype == kDefaultStorage && rhs_stype == kCSRStorage &&
```

Comment:
Is the implementation only available on CPU? No fallback on GPU ctx?

Reply:
I have added a check for CPU. will fallback to default storage for gpu